tests/test_hf_api.py

```diff
@@ -2449,6 +2449,41 @@ def test_request_space_hardware(self, post_mock: Mock) -> None:
         )
 
 
+class ListRepoRefsTest(unittest.TestCase):
+    @classmethod
+    @with_production_testing
```

Comment:
Smart to only define it on the `setUpClass` for it to act on all subsequent methods. I wonder if it wouldn't be cleaner to put it on the class itself so that it's very clear that it should affect the entire class?

Reply:
Actually I did that as a workaround because 
```py
@with_production_testing
class ListRepoRefsTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
```
was not working for me :cry: 
But I have to admit my debug time on that was approximately 30s, just the time to try some possibilities and take the first one that worked :roll_eyes:

Reply:
Haha in that case we can also keep it there, it was just a nit, to be fair.